kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyInitializer.java

```diff
@@ -174,7 +176,7 @@ protected void onLookupComplete(ChannelHandlerContext ctx, Future<SslContext> fu
 
     @VisibleForTesting
     void addHandlers(SocketChannel ch, VirtualClusterBinding binding) {
-        var virtualCluster = binding.virtualClusterModel();
+        var virtualCluster = ((VirtualClusterModel) binding.endpointListener());
```

Comment:
Wondering if we should add `binding.endpointListener().virtualCluster()` here, vs adding more methods onto the Listener. I see we are already hiding that they share a target cluster with:

```
 @Override
        public TargetCluster targetCluster() {
            return virtualCluster.targetCluster();
        }
```

Or should we make the Listener the thing that gets passed everywhere, and you ask the Listener for the filters and virtual cluster name?

Reply:
I see you addressed that.